app/src/ui/cherry-pick/cherry-pick-conflicts-dialog.tsx

```diff
@@ -123,14 +123,19 @@ export class CherryPickConflictsDialog extends React.Component<
       repository,
       dispatcher,
       step,
-      sourceBranchName: ourBranch,
+      sourceBranchName,
     } = this.props
 
     const {
       manualResolutions,
       targetBranchName: theirBranch,
     } = step.conflictState
 
+    let ourBranch: string
+    if (sourceBranchName !== null) {
+      ourBranch = sourceBranchName
+    }
```

Comment:
Doesn't this code make the app fail when `ourBranch` is used a few lines below? Because if `sourceBranchName` happens to be `null`, `ourBranch` would be uninitialized 🤔 

![image](https://user-images.githubusercontent.com/1083228/109838912-e109c180-7c46-11eb-8a04-50ade5a784b7.png)

Reply:
No. Actually and you can test this scenario too! Because it will be null if you start a conflicted cherry pick outside of our app. The `renderUnmergedFile` is made to gracefully handle it .. maybe not the most optimal, but you do not get the name of the source on that dropdown for which side of the conflict you want in the conflict dialog.. but you still get the target branch name for opposing.. thus, in this unique scenario you can still differentiate between target and source even if not in the most clear manner.

I looked into what it would take to determine the source branch in this scenario... and it is technically possible and I have in my cherry picking notes to look into it further to make this one scenario better. But, I was comfortable as this is usable and there are paths that fail much harder in cherry picking that I think warrant more attention (starting without a clean working tree for example).

Reply:
![image](https://user-images.githubusercontent.com/75402236/109859831-68e1d280-7c2b-11eb-81cf-7cc8b48195b5.png)

Reply:
I'm sorry, I wasn't referring at the possibility of not having a source branch. You explained that scenario perfectly in one of the comments (and I'm very grateful for that 🙏 ❤️ ).

In this case I'm referring strictly to the code side of things:

```
    let ourBranch: string // declared, but not initialized!!!
    if (sourceBranchName !== null) {
      ourBranch = sourceBranchName // initialized 😌 
    }
    // what happens at this point if the sourceBranchName was null?
    // ourBranch would still be uninitialized, which should fail at compile time as I showed in my screenshot, no?
```

I will check out this branch later and look at this 👆  if I have time before you get back to it 😅

Reply:
Yes. This was intentional. In fact, I just updated the code to show that intention better.

As you can see below, the function `renderUnmergedFIle` is the only place I use the `ourBranch` and on the other side of the split you can see that `ourBranch` has the ? as an optional property. If I send in `undefined`, it will be same as if I did not send it at all. This is what I want, I want to send it if the sourceBranch exists, but I don't want to send it if I don't have anything to send.

![image](https://user-images.githubusercontent.com/75402236/109952583-fdd8e000-7cac-11eb-9510-179dfbf02769.png)
![image](https://user-images.githubusercontent.com/75402236/109954198-0c27fb80-7caf-11eb-8f41-e944bcc3f091.png)

Reply:
Ohhh!! I totally missed that `?`, but the new code makes it even clearer, thank you!